### PR TITLE
regular_file.c: don't treat 0 as a bad file descriptor

### PR DESCRIPTION
**Pull request by sporksmith**
I wanted to avoid comparing with equality to OSFILE_INVALID everywhere, in case we ever want to add other negative error values. Added an inline helper `_fd_isValid` instead.

Most of these checks could probably be omitted altogether in favor of letting the native file operation fail and return EBADF for us or us later failing to find the fd in our descriptor table, but probably not worth changing it right now.

Fixes https://github.com/shadow/shadow/issues/2604

### CODE REVIEW

File: src/main/host/descriptor/regular_file.c [hunk 1]
```diff
@@ -531,11 +531,11 @@ ssize_t regularfile_preadv2(RegularFile* file, const Host* host, const struct io
 ssize_t regularfile_write(RegularFile* file, const void* buf, size_t bufSize) {
     MAGIC_ASSERT(file);
 
-    if (!_fd_isValid(_regularfile_getOSBackedFD(file))) {
+    if (file->type == FILE_TYPE_IN_MEMORY) {
         return -EBADF;
     }
 
-    if (file->type == FILE_TYPE_IN_MEMORY) {
+    if (!_fd_isValid(_regularfile_getOSBackedFD(file))) {
```
Comment (robgjansen) [inline src/main/host/descriptor/regular_file.c:538]: I don't understand why this change is necessary, or the other similar changes below, because I think the code is equivalent with or without the change?

Since the two `if` statements are executed in sequence, isn't it equivalent to

```C
if (file->type == FILE_TYPE_IN_MEMORY || !_fd_isValid(_regularfile_getOSBackedFD(file))) {
    return -EBADF;
}    
```

in which case the order should not matter?

File: src/main/host/descriptor/regular_file.c [hunk 2]
```diff
@@ -888,6 +884,9 @@ int regularfile_fcntl(RegularFile* file, unsigned long command, void* arg) {
         arg = (void*)arg_int;
     }
 
+    if (!_fd_isValid(_regularfile_getOSBackedFD(file))) {
+        return -EBADF;
+    }
```
Comment (robgjansen) [inline src/main/host/descriptor/regular_file.c:889]: Above we are potentially making a change to some file state in Shadow, and returning some values in the arg, but then returning an `EBADF` error. But if we return an error, doesn't that mean that we don't want to make any changes to shadow state (and not write to arg memory) based on this erroneous call?

If this _is_ still a valid path, then a comment to explain why would be useful.

**Review by robgjansen — APPROVED** [review]

File: src/main/host/descriptor/regular_file.c (hunk 1 above)
Comment (sporksmith) [reply src/main/host/descriptor/regular_file.c:538]: Yup! Basically I was moving the `_isValid` checks to just-before the native calls, to better convey/preserve the intent that it's there primarily to avoid a native syscall that we know would fail.

File: src/main/host/descriptor/regular_file.c (hunk 2 above)
Comment (sporksmith) [reply src/main/host/descriptor/regular_file.c]: Yeah, this one is a bit of a grey area. Reverted this one.
